src/bots/ama/subCommands/editEvent.js

```diff
@@ -0,0 +1,83 @@
+import EventModel from '../models/Event';
+import client from '../../../client';
+import parseObject from '../parser';
+import {
+  getFormattedEvent,
+  getStrings,
+  possibleEditFields
+} from '../constants';
+
+/**
+ * Handles editing an event with Event schema and sends message with updated event
+ * @param {Array.<string>} args - rest of command arguments
+ */
+const handler = async (args) => {
+  if (args.length < 1) {
+    client.message.channel.send(
+      getStrings(possibleEditFields).editEventExample
+    );
+    return;
+  }
+
+  let targetEvent;
+  let targetId = args[0];
+
+  try {
+    targetEvent = await EventModel.findById(targetId);
+  } catch {
+    client.logger.debug('invalid ObjectId type supplied');
+  }
+
+  if (targetEvent) {
+    let targetBlock = client.message.content.substring(
+      client.message.content.indexOf(targetId) + targetId.length + 1
+    );
+
+    let tokens = tokenizeEvent(targetBlock);
+
+    targetEvent = await EventModel.findOneAndUpdate(
+      { _id: targetId },
+      { $set: tokens },
+      { new: true }
+    );
+
+    client.message.channel.send(getFormattedEvent(targetEvent, true));
+  } else {
+    client.message.channel.send(getStrings(targetId).eventNotFound);
+  }
+};
+
+/**
+ * Tokenizes a string block by splitting on double newline then keys are values pre colon and values past colon
+ * @param {string} block - block of text to be turned into plain javascript object
+ */
+let tokenizeEvent = (block) => {
+  let split = block.split('\n\n');
```

Comment:
just thinking about how usable this is, do we really want to require users to space it out like this? seems like it could be prone to user error. doesn't have to be now, but maybe we should think of a better solution in the future, whether that's putting in answers to a form one field at a time like some bots do or some syntax that's more clear for users.

Reply:
since it's just mods using it, it's not super important though so can make it a low priority issue

Reply:
Going to mark this as resolved and created issue in accordance https://github.com/cscareerhub/megabot/issues/75